scripts/picard.in

```diff
@@ -1,3 +1,30 @@
 #!/usr/bin/env python3
-from picard.tagger import main
-main('%(localedir)s', %(autoupdate)s)
+
+try:
+    from picard.tagger import main
+    main('%(localedir)s', %(autoupdate)s)
+except SystemExit:
+    raise  # Just continue with a normal application exit
+except:  # noqa: F722
+    # First try to get traceback information and write it to a log file
+    # with minimum chance to fail.
+    import os
+    import sys
+    import tempfile
+    import traceback
+    trace = traceback.format_exc()
+    logfile = None
+    try:
+        (fd, logfile) = tempfile.mkstemp(".log", "picard-crash-")
+        os.write(fd, trace.encode(errors="replace"))
```

Comment:
Is there a reason to not use `NamedTemporaryFile()` together with `with` here?

```python
with NamedTemporaryFile(suffix='.log', prefix='picard-crash-', delete=False) as f:
   f.write(trace.encode(errors="replace"))
   logfile = f.name
```

Reply:
I mainly did this because the file must not be deleted. But I see know using NamedTemporaryFile with delete=False achieves this as well.